yhat/deployment/save_session.py

```diff
@@ -299,11 +308,14 @@ def _spider_function(function, session, pickles={}):
             elif _is_spark(obj):
                 logging.debug("\t%s is from spark. serializing using `dumps_spark_to_base64`." % varname)
                 pickles[varname] = terragon.dumps_spark_to_base64(session['sc'], obj)
+            elif _is_pom(obj):
+                logging.debug("\t%s is from pomegranate. serializing using `dumps_pom_to_base64`." % varname)
+                pickles[varname] = terragon.dumps_pom_to_base64(obj)
             else:
                 logging.debug("\tno special serialization requirement detected for %s. using dumps_to_base64." % varname)
                 pickles[varname] = terragon.dumps_to_base64(obj)
 
-        if hasattr(obj, "__module__"):
+        elif hasattr(obj, "__module__"):
             logging.debug("\tobject %s has attribute __module__" % varname)
```

Comment:
@glamp unless I'm missing something, I think this should be an `elif`. It seems like if we serialize the object under the first condition of this big if (`if hasattr(obj, "__name__")==False:`) then we wouldn't want to start a new if sequence? The pomegranate objects were getting serialized with the first case, but then getting serialized again with the pickle catch all on line 359.  Seemed strange...